### PR TITLE
feat(perf-issues): Slow db span detection only on queries starting with `SELECT`

### PR DESCRIPTION
**Pull request by 0Calories**
Changes the slow db span detector to only look at queries that start with `SELECT` and updates the tests to reflect this change.

### CODE REVIEW

File: fixtures/events/performance_problems/n-plus-one-in-django-index-view.json [hunk 1]
```diff
@@ -168,7 +168,7 @@
       "same_process_as_parent": true
     },
     {
-      "timestamp": 1661957871.418989,
+      "timestamp": 1661959871.418989,
```
Comment (gggritso) [inline fixtures/events/performance_problems/n-plus-one-in-django-index-view.json:171]: Why the timestamp change here?

Comment (0Calories) [reply fixtures/events/performance_problems/n-plus-one-in-django-index-view.json:171]: That span wasn't slow enough to be detected by the slow span detector

**Review by gggritso — APPROVED** [review]

**Comment by 0Calories** [comment]
@gggritso Agreed, but the reason why I opted to edit the N+1 fixture is specifically because of this test:

https://github.com/getsentry/sentry/blob/f5bb08c36e01c9ede0ffa6c2388c36c67910721a/tests/sentry/utils/performance_issues/test_performance_detection.py#L637

It's testing that 5 diff issues are detected, including slow span, and before it was working due to an `http` span being slow in the fixture.

**Comment by 0Calories** [comment]
I guess I can address this in a later PR, since I had plans of cleaning up some of the tests here!

**Pull request merged**